### PR TITLE
Release/0.40.0

### PR DESCRIPTION
**Pull request by samchrisinger**

### CODE REVIEW

File: framework/tasks/handlers.py [hunk 1]
```diff
@@ -48,11 +52,8 @@ def queued_task(task):
     """
     @functools.wraps(task)
     def wrapped(*args, **kwargs):
-        if settings.USE_CELERY:
```
Comment (sloria) [inline framework/tasks/handlers.py]: Why was this removed?

Comment (samchrisinger) [inline framework/tasks/handlers.py]: Moved to the teardown_request.

This move the USE_CELERY logic to a much more broadly applicable place.

Comment (sloria) [inline framework/tasks/handlers.py]: Won't this cause an error if USE_CELERY is `False`?

Comment (samchrisinger) [inline framework/tasks/handlers.py]: Shouldn't. Docstring says can only be applied to celery tasks, which all expose a .si method. Any task can be called inline like ```TASK.si(*args, **kwargs)()```

Comment (sloria) [inline framework/tasks/handlers.py]: Cool; thanks for clarifying
